src/graph/SpatialMergePointsNode.cpp

```diff
@@ -31,12 +31,13 @@ void SpatialMergePointsNode::setParameters(const std::vector<rgl_field_t>& field
 	}
 }
 
-void SpatialMergePointsNode::validate()
+void SpatialMergePointsNode::onInputChange()
 {
 	pointInputs = Node::filter<IPointsNode>(this->inputs);
 
-	if (pointInputs.empty()) {
-		auto msg = "expected at least one IPointsNode input for SpatialMergePointsNode";
+	if (pointInputs.size() < 2) {
+		// Having just one input does not make sense, because it is equivalent to removing this Node.
```

Comment:
I would stop right there for a while.
In the case of single input, this node is just a pass-through. This is not a mistake, just useless node.
I can imagine situation when we create a graph with spatial merge inside, only one input. This graph is working for some time, and when the situation is ready, we add second input (node or subgraph) to an already created and working graph.
This is not a no-go, however, re-think it.

Reply:
So, why not add SpatialMerge when it's actually needed? Having it operate on just 1 input smells very much like a bug  / unintended situation.

Reply:
So, why not add SpatialMerge when it's actually needed? Having it operate on just 1 input smells very much like a bug  / unintended situation.

Reply:
I partially agree with You.
I would warn the user about it, but still let it.

Reply:
I get your reasoning; also, consider the fact that the warning would land in log file and / or stdout, which requires additional effort to see in a typical RGL usage scenario (engine plugins). Therefore, I prefer an error, which is a minor inconvenience if the passthrough was desired and a huge helper if it was not.